# origin: Debian (herbert)
# cset: n/a
# inclusion: not suitable for mainline
# description: add an option to disable ACPI by default
# revision date: 2004-09-05

--- a/Documentation/Configure.help   2004-04-14 23:05:24.000000000 +1000
+++ b/Documentation/Configure.help	2004-04-17 14:23:30.000000000 +1000
@@ -20029,6 +20195,11 @@
   ACPI C2 and C3 processor states to save power, on systems that
   support it.
 
+Disable ACPI by default
+CONFIG_ACPI_OFF
+  If you say `Y' here, then ACPI support will be disabled unless you
+  pass the "acpi=on" option to the kernel at boot time.
+
 ACPI AC Adapter
 CONFIG_ACPI_AC
   This driver adds support for the AC Adapter object, which indicates
diff -urN a/drivers/acpi/Config.in b/drivers/acpi/Config.in
--- a/drivers/acpi/Config.in	2004-04-14 23:05:28.000000000 +1000
+++ b/drivers/acpi/Config.in	2004-04-17 14:23:44.000000000 +1000
@@ -19,6 +19,9 @@
     fi
     define_bool CONFIG_ACPI_SLEEP		y
     define_bool CONFIG_ACPI_SYSTEM		y
+    if [ "$CONFIG_X86" = "y" ]; then
+      bool         '  Off By Default' CONFIG_ACPI_OFF
+    fi
     tristate     '  AC Adapter'	CONFIG_ACPI_AC
     tristate     '  Battery'		CONFIG_ACPI_BATTERY
     tristate     '  Button'		CONFIG_ACPI_BUTTON
diff -urN a/arch/i386/kernel/setup.c b/arch/i386/kernel/setup.c
--- a/arch/i386/kernel/setup.c	2004-04-14 23:05:25.000000000 +1000
+++ b/arch/i386/kernel/setup.c	2004-04-17 14:23:37.000000000 +1000
@@ -177,7 +177,7 @@
 static int disable_x86_serial_nr __initdata = 1;
 static u32 disabled_x86_caps[NCAPINTS] __initdata = { 0 };
 
-#ifdef	CONFIG_ACPI_INTERPRETER
+#if defined(CONFIG_ACPI_INTERPRETER) && !defined(CONFIG_ACPI_OFF)
 	int acpi_disabled = 0;
 #else
 	int acpi_disabled = 1;
@@ -845,6 +845,12 @@
 #endif
 
 #ifdef CONFIG_ACPI_BOOT
+		/* "acpi=on" enables ACPI table parsing and interpreter */
+		else if (!memcmp(from, "acpi=on", 7)) {
+			acpi_ht = 1;
+			acpi_disabled = 0;
+		}
+
 		/* "acpi=off" disables both ACPI table parsing and interpreter */
 		else if (!memcmp(from, "acpi=off", 8)) {
 			disable_acpi();
